posts/2024-12-10_why-gno/README.md

```diff
@@ -0,0 +1,102 @@
+
+The technical innovation brought by Gno makes solving the above problems possible.
+
+- We're at the Assembly age of web3.
+    - Developing software is costly:
+      - It's expensive to build basic things, both in terms of expertise in development, but also auditing and finding people that truly understand and can take advantage of the technology.
+      - Many ecosystems employ domain-specific languages or restrictions on top of others (ie. WASM), which still require specific training.
+  - The development which is possible is very basic, similar to "interconnected spreadsheets".
+  - Monolithic blockchains create limitations which de-facto limit the growth of the chain, only allowing for the entire system a very limited amount of computation in each block.
+- What Solidity misses:
+  - Solidity is a Domain Specific Language, further increasing the cost of adoption by new users, especially those who are still curious and uncommitted to working on a blockchain.
+  - Financially focused. Solidity contains many native constructs pertaining specifically to smart contracts; like the global `ether`, the type `address`, the globals `msg`, `tx` and `block`. On the other hand, its design lacks simple ways to represent and handle floats, dates and times, and misses other general-purpose features programmers expect from most programming languages.
+  - Hidden control flow: [fallback functions](https://docs.soliditylang.org/en/latest/contracts.html#fallback-function) create hidden control flow when sending tokens, causing problems like [re-entrancy attacks](https://docs.soliditylang.org/en/latest/security-considerations.html#reentrancy).
+- Why Gno solves it:
+  - Composability:
+      - Interfaces and closures allow to plug different code and data structures into each other.
+      - Realms can be composed re-using functionality of other realms or existing functionality published as packages.
+      - IBC enables the blockchains using Gno to communicate with one another seamlessly, and build on the work happening elsewhere.
+  - Low cost of adoption: Gno is a subset of Go with some different standard libraries. Learning Gno represents a smaller cost of adoption, as the skill can be widely used outside of gno.land; and it also a language that is used by [around two million developers worldwide](https://research.swtch.com/gophercount).
+  - Low cost of building: quicker and simpler to write dapps (simple language, good already-audited p/ libraries); quicker and cheaper to audit dapps (all the source code is to be published on-chain).
+  - App-focused: we are interested in social platforms, video game servers, and ways to share knowledge in a decentralised manner. We see dApps as systems that can make the web decentralised again and create a truly open internet for the time to come. De-fi is great - and an obvious usecase for blockchains - Gno expands on what de-fi can do and enable many more apps, and ways to communicate with others.
```

Comment:
```suggestion
  - App-focused: we are interested in social platforms, video game servers, and ways to share knowledge in a decentralised manner. We see dApps as systems that can make the web decentralised again and create a truly open internet for the time to come. DeFi is great - and an obvious usecase for blockchains - Gno expands on what DeFi can do and enable many more apps, and ways to communicate with others.
```

Reply:
It's worth calling out inconsistent use of the term `dapps` and `dApps` as well